Source/Core/Gen5/Generators/IDGenerator5.cpp

```diff
@@ -50,19 +49,13 @@ std::vector<IDState5> IDGenerator5::generate(u64 seed, u32 pid, bool checkPID)
 
         if (filter.compare(state))
         {
-            if (checkPID)
-            {
-                bool idBit = (tid & 1) ^ (sid & 1);
+            bool isShiny = (psv >> 3) == state.getTSV();
 
-                // Check if PID will be modified by the tid/sid combo
-                u16 actualPSV = (idBit ^ pidBit) ? xorPSV : psv;
-                if ((actualPSV >> 3) == state.getTSV())
-                {
-                    state.setSeed(seed);
-                    states.emplace_back(state);
-                }
-            }
-            else
+            // Check if PID is possible with TID/SID combo if Wild/Stationary box is checked
+            if (isShiny && checkXOR) // We need to do the check only if it was shiny first
+                isShiny = ((tid & 1) ^ (sid & 1)) == pidBit;
```

Comment:
I literally have no clue what this is doing. Why was it changed from what zak and I agreed on

Also make sure if statements have opening and closing brackets